### ISSUE TITLE
PDF file upload to AWS S3 doesn't work on AWS SDK v3, but works on AWS SDK v2

### ISSUE THREAD

**Issue by MichaelGradek** [oven-sh/bun#4204]
### What version of Bun is running?

0.7.3

### What platform is your computer?

Darwin 22.2.0 arm64 arm

### What steps can reproduce the bug?

Upload a PDF file to the following endpoint using POST multipart/data under a field named "file". The API is powered by Elysia framework, this is the package.json file:

```json
{
  "name": "invoices",
  "version": "1.0.48",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "dev": "bun run --watch src/index.ts"
  },
  "dependencies": {
    "@aws-sdk/client-s3": "^3.388.0",
    "@aws-sdk/lib-storage": "^3.388.0",
    "aws-sdk": "^2.1435.0",
    "axios": "^1.4.0",
    "elysia": "latest", // currently on 0.6.3
    "nanoid": "^4.0.2",
    "parse-multipart-data": "^1.5.0",
    "pino": "^8.15.0"
  },
  "devDependencies": {
    "bun-types": "latest",
    "pino-pretty": "^10.2.0"
  },
  "module": "src/index.js"
}
```

... and here's the controller:

```typescript
import {Elysia, t} from 'elysia'
import multipart from 'parse-multipart-data'
import {Upload} from "@aws-sdk/lib-storage" // SDK v3
import {S3Client} from "@aws-sdk/client-s3" // SDK v3
import {S3} from 'aws-sdk' // SDK v2


export const UploadController = (app: Elysia) => {
    app.group('/upload', (app) =>
        app.post('/', async (ctx) => {
            const rawBody = Buffer.from(await ctx.request.arrayBuffer());
            const boundary = ctx.headers["content-type"].split("boundary=")[1];
            const parts = multipart.parse(rawBody, boundary);
            const file = parts[0].data
            const mimeType = parts[0].type
            for (const p of parts) {
                console.log(p.filename)
                console.log(p.type)

                const params = {
                    Key : p.filename!,
                    Body : p.data,
                    Bucket : 'foo-bar-bucket',
                    ContentType : p.type
                }

                const sdkv2Client = new S3()
                const sdkv3Client = new S3Client()

                // SDK V2 Upload works for images and PDF files, all good here, the only thing is V2 SDK will
                // be in maintenance mode, obviously you'd want to migrate to SDK V3
                try {
                    const s3Upload = await sdkv2Client.putObject(params).promise()
                    console.log(s3Upload)
                } catch (e) {
                    console.error(e)
                }

                // SDK V3 Upload ONLY works for IMAGES, not PDF files
                try {
                    const parallelUploads3 = new Upload({
                        client: sdkv3Client,
                        params: params
                    });

                    parallelUploads3.on('httpUploadProgress', (progress) => {
                        console.log(progress)
                    })

                    const done = await parallelUploads3.done();
                    console.log("DONE!")
                    console.log(done)
                } catch (e) {
                    console.error("ERROR");
                    console.error(e);
                }
            }

            return new Response(file, { status: 202, headers: { 'Content-Type': mimeType } });
        })
    )

    return app
}
```

### What is the expected behavior?

The PDF file can be seen in the S3 bucket

### What do you see instead?

No PDF file can be found in the S3 Bucket

### Additional information

If, however, you send an image through, the image DOES upload correctly onto S3 on both SDK versions, so it seems like it's only affecting PDF files (and potentially other file formats I haven't tried?)

**Comment by ymc182** [comment]
I am having a similar problem, the status return 100 instead of 200, smaller files (eg. under 50kb) uploaded to the bucket but anything larger then 500kb couldn't upload.

**Comment by MichaelGradek** [comment]
This is now resolved as of version 1.0.6 (probably earlier versions fixed this, I'm guessing 1.0.1)

**Comment by impin2rex** [comment]
> ### What version of Bun is running?
> 0.7.3
> 
> ### What platform is your computer?
> Darwin 22.2.0 arm64 arm
> 
> ### What steps can reproduce the bug?
> Upload a PDF file to the following endpoint using POST multipart/data under a field named "file". The API is powered by Elysia framework, this is the package.json file:
> 
> ```json
> {
>   "name": "invoices",
>   "version": "1.0.48",
>   "scripts": {
>     "test": "echo \"Error: no test specified\" && exit 1",
>     "dev": "bun run --watch src/index.ts"
>   },
>   "dependencies": {
>     "@aws-sdk/client-s3": "^3.388.0",
>     "@aws-sdk/lib-storage": "^3.388.0",
>     "aws-sdk": "^2.1435.0",
>     "axios": "^1.4.0",
>     "elysia": "latest", // currently on 0.6.3
>     "nanoid": "^4.0.2",
>     "parse-multipart-data": "^1.5.0",
>     "pino": "^8.15.0"
>   },
>   "devDependencies": {
>     "bun-types": "latest",
>     "pino-pretty": "^10.2.0"
>   },
>   "module": "src/index.js"
> }
> ```
> 
> ... and here's the controller:
> 
> ```ts
> import {Elysia, t} from 'elysia'
> import multipart from 'parse-multipart-data'
> import {Upload} from "@aws-sdk/lib-storage" // SDK v3
> import {S3Client} from "@aws-sdk/client-s3" // SDK v3
> import {S3} from 'aws-sdk' // SDK v2
> 
> 
> export const UploadController = (app: Elysia) => {
>     app.group('/upload', (app) =>
>         app.post('/', async (ctx) => {
>             const rawBody = Buffer.from(await ctx.request.arrayBuffer());
>             const boundary = ctx.headers["content-type"].split("boundary=")[1];
>             const parts = multipart.parse(rawBody, boundary);
>             const file = parts[0].data
>             const mimeType = parts[0].type
>             for (const p of parts) {
>                 console.log(p.filename)
>                 console.log(p.type)
> 
>                 const params = {
>                     Key : p.filename!,
>                     Body : p.data,
>                     Bucket : 'foo-bar-bucket',
>                     ContentType : p.type
>                 }
> 
>                 const sdkv2Client = new S3()
>                 const sdkv3Client = new S3Client()
> 
>                 // SDK V2 Upload works for images and PDF files, all good here, the only thing is V2 SDK will
>                 // be in maintenance mode, obviously you'd want to migrate to SDK V3
>                 try {
>                     const s3Upload = await sdkv2Client.putObject(params).promise()
>                     console.log(s3Upload)
>                 } catch (e) {
>                     console.error(e)
>                 }
> 
>                 // SDK V3 Upload ONLY works for IMAGES, not PDF files
>                 try {
>                     const parallelUploads3 = new Upload({
>                         client: sdkv3Client,
>                         params: params
>                     });
> 
>                     parallelUploads3.on('httpUploadProgress', (progress) => {
>                         console.log(progress)
>                     })
> 
>                     const done = await parallelUploads3.done();
>                     console.log("DONE!")
>                     console.log(done)
>                 } catch (e) {
>                     console.error("ERROR");
>                     console.error(e);
>                 }
>             }
> 
>             return new Response(file, { status: 202, headers: { 'Content-Type': mimeType } });
>         })
>     )
> 
>     return app
> }
> ```
> 
> ### What is the expected behavior?
> The PDF file can be seen in the S3 bucket
> 
> ### What do you see instead?
> No PDF file can be found in the S3 Bucket
> 
> ### Additional information
> If, however, you send an image through, the image DOES upload correctly onto S3 on both SDK versions, so it seems like it's only affecting PDF files (and potentially other file formats I haven't tried?)

Somehow, sdkV3 isn't working for me as well.
